build-aux/snap/snapcraft.yaml

```diff
@@ -160,3 +161,30 @@ parts:
       cp -a fc-cache-bionic $SNAPCRAFT_PART_INSTALL/bin/fc-cache-v7
     prime:
       - bin/fc-cache-v7
+  apparmor:
+    plugin: autotools
+    build-packages: [bison, flex, gettext]
+    source: https://launchpad.net/apparmor/3.0/3.0.7/+download/apparmor-3.0.7.tar.gz
+    override-build: |
+      cd $SNAPCRAFT_PART_BUILD/libraries/libapparmor
+      ./autogen.sh
+      ./configure --prefix=/usr --disable-man-pages --disable-perl --disable-python --disable-ruby
+      make -j
+      # place libapparmor into staging area for use by snap-confine
+      make -C src install DESTDIR=$SNAPCRAFT_STAGE
+      cd $SNAPCRAFT_PART_BUILD/parser
+      # copy in a pregenerated list of network address families so that the
+      # parser gets built to support as many as possible even if glibc in
+      # the current build environment does not support them
+      cp $SNAPCRAFT_PROJECT_DIR/build-aux/snap/apparmor/af_names.h .
+      make -j
+      mkdir -p $SNAPCRAFT_PART_INSTALL/usr/lib/snapd
+      cp -a apparmor_parser $SNAPCRAFT_PART_INSTALL/usr/lib/snapd/
+      mkdir -p $SNAPCRAFT_PART_INSTALL/usr/lib/snapd/apparmor
+      cp -a parser.conf $SNAPCRAFT_PART_INSTALL/usr/lib/snapd/apparmor/
+      cd $SNAPCRAFT_PART_BUILD/profiles
+      make -j
+      mkdir -p $SNAPCRAFT_PART_INSTALL/usr/lib/snapd/apparmor.d
+      cp -a apparmor.d/abi $SNAPCRAFT_PART_INSTALL/usr/lib/snapd/apparmor.d/
+      cp -a apparmor.d/abstractions $SNAPCRAFT_PART_INSTALL/usr/lib/snapd/apparmor.d/
+      cp -a apparmor.d/tunables $SNAPCRAFT_PART_INSTALL/usr/lib/snapd/apparmor.d/
```

Comment:
should the build spread test explicitly or does it already implicitly test this various copying?

Reply:
This is implicitly tested in the snapd-snap spread test via the line: https://github.com/snapcore/snapd/pull/12087/files#diff-91d3eb78b80e15bf4e9a9f5dfe17fdda2a1e07cbb574e0113d6bea9c6bb8fb85R256